lib/Recur/RRuleIterator.php

```diff
@@ -127,55 +127,120 @@ public function isInfinite()
-                        $i = $diff->days * 24;
-                        break;
-                    case 'daily':
-                        $i = $diff->days;
-                        break;
-                    case 'weekly':
-                        $i = $diff->days / 7;
-                        break;
-                    case 'monthly':
-                        $i = $diff->days / 30;
-                        break;
-                    case 'yearly':
-                        $i = $diff->days / 365;
-                        break;
-                }
-                $i /= $this->interval;
-                $i /= 4;
-                $i = floor($i);
-                $i = max(1, $i);
-                do {
-                    $previousDate = clone $this->currentDate;
-                    $this->next($i);
-                } while ($this->valid() && $this->currentDate < $dt);
-
-                $this->currentDate = $previousDate;
-                // do one step to avoid deadlock
-                $this->next();
-            } while ($i > 5 && $this->valid() && $this->currentDate < $dt);
+            $this->jumpForward($dt);
         }
 
         while ($this->valid() && $this->currentDate < $dt) {
             $this->next();
         }
+    }
 
+    /**
+     * This method allows you to quickly go to the next occurrence before the specified date.
+     *
+     * @param DateTimeInterface $dt
+     */
+    public function fastForwardBefore(DateTimeInterface $dt)
+    {
+        // We don't do any jumps if we have a count limit as we have to keep track of the number of occurrences
         if (!isset($this->count)) {
-            // We don't know the counter at this point anymore
-            $this->counter = NAN;
+            $this->jumpForward($dt);
+        }
+
+        $previousDate = null;
+        while ($this->valid() && $this->currentDate < $dt) {
+            $previousDate = clone $this->currentDate;
+            $this->next();
+        }
+
+        isset($previousDate) && $this->currentDate = $previousDate;
+    }
+
+    /**
+     * Return the frequency in number of days.
+     *
+     * @return float|int|null
+     */
+    private function getFrequencyCoeff()
+    {
+        $frequencyCoeff = null;
+
+        switch ($this->frequency) {
+            case 'hourly':
+                $frequencyCoeff = 1 / 24;
+                break;
+            case 'daily':
+                $frequencyCoeff = 1;
+                break;
+            case 'weekly':
+                $frequencyCoeff = 7;
+                break;
+            case 'monthly':
+                $frequencyCoeff = 30;
+                break;
+            case 'yearly':
+                $frequencyCoeff = 365;
+                break;
+        }
+
+        return $frequencyCoeff;
+    }
+
+    /**
+     * Perform a fast forward by doing jumps based on the distance of the requested date and the frequency of the
+     * recurrence rule. Will return the last occurrence before the requested date.
```

Comment:
I don't see a return statement, can you update the doc please?

Reply:
Indeed, it's corrected! Thanks